webapp/tests/Utils/Scoreboard/ScoreboardIntegrationTest.php

```diff
@@ -0,0 +1,281 @@
+<?php declare(strict_types=1);
+
+namespace Tests\Utils\Scoreboard;
+
+use App\Entity\Contest;
+use App\Entity\ContestProblem;
+use App\Entity\Judgehost;
+use App\Entity\Judging;
+use App\Entity\Language;
+use App\Entity\Problem;
+use App\Entity\Submission;
+use App\Entity\Team;
+use App\Entity\TeamCategory;
+use App\Service\DOMJudgeService;
+use App\Service\ScoreboardService;
+use App\Utils\Scoreboard\Scoreboard;
+use App\Utils\Scoreboard\TeamScore;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class ScoreboardTest extends KernelTestCase
+{
+    public const CONTEST_NAME = 'scoretest';
+    public const NUM_PROBLEMS = 3;
+    public const NUM_TEAMS = 3;
+
+    /**
+     * @var \App\Service\DOMJudgeService
+     */
+    private $dj;
+
+    /**
+     * @var \App\Service\ScoreboardService
+     */
+    private $ss;
+
+    /**
+     * @var \Doctrine\ORM\EntityManager
+     */
+    private $em;
+
+    /**
+     * @var App\Entity\Contest
+     */
+    private $contest;
+
+    /**
+     * @var App\Entity\Judgehost
+     */
+    private $judgehost;
+
+    /**
+     * @var App\Entity\Problem[]
+     */
+    private $problems;
+
+    /**
+     * @var App\Entity\Team[]
+     */
+    private $teams;
+
+    protected function setUp()
+    {
+        self::bootKernel();
+
+        $this->dj = self::$container->get(DOMJudgeService::class);
+        $this->ss = self::$container->get(ScoreboardService::class);
+        $this->em = self::$container->get('doctrine')->getManager();
+
+        // Create a contest, problems and teams for which to test the
+        // scoreboard. These get deleted again in tearDown().
+        $this->contest = new Contest();
+        $this->contest
+            ->setExternalid(self::CONTEST_NAME)
+            ->setName(self::CONTEST_NAME)
+            ->setShortname(self::CONTEST_NAME)
+            ->setStarttimeString(date('Y').'-01-01 10:00:00 Europe/Amsterdam')
```

Comment:
Does this mean the test will fail between 0:00 and 10:00 on the first of January?

Reply:
That's a great idea to prevent us to submit/merge anything while we are drunk :beers:

Reply:
Hmmm, maybe so. The submissions are automatically marked as judged, so I don't really know what would happen in that small time window. But I can change it to Y-1.

Reply:
This will most likely bite me in the next "happy new year" commit